### PR TITLE
bump code gen v0.14.1 / aws sdk to 1.37.10

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
bump code gen v0.14.1 and aws sdk to 1.37.10

#### Breaking API Changes
**cloudfront distribution  (originAccessIDentity -> originAccessIdentity)**
**lambda function removed Layers in CreateFunctionInput** in general we have the following comment: https://github.com/crossplane/provider-aws/blob/master/pkg/controller/lambda/function/setup.go#L145-L149

Fixes:
Generate typedefs for payload fields in Spec and Status:  https://github.com/aws-controllers-k8s/code-generator/pull/146
aws-sdk-go version bump, 1.37.4 -> 1.37.10: https://github.com/aws-controllers-k8s/code-generator/pull/143
https://github.com/crossplane/provider-aws/pull/543
apigatewayv2 bump to 1.37.10 https://github.com/aws-controllers-k8s/community/issues/951

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

make generate
make reviewable

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by haarchri** [comment]
locally make check-diff is working:

```
➜  provider-aws git:(code-gen-v0.9.2) make check-diff
HEAD ist jetzt bei 4d0db1b Update to ACK Runtime `v0.9.2` (#152)
22:52:04 [ .. ] Generating apigatewayv2 controllers and CRDs
22:52:36 [ OK ] Generating apigatewayv2 controllers and CRDs
22:52:36 [ .. ] Generating cloudfront controllers and CRDs
22:52:55 [ OK ] Generating cloudfront controllers and CRDs
22:52:55 [ .. ] Generating dynamodb controllers and CRDs
22:53:13 [ OK ] Generating dynamodb controllers and CRDs
22:53:13 [ .. ] Generating efs controllers and CRDs
22:53:51 [ OK ] Generating efs controllers and CRDs
22:53:51 [ .. ] Generating kafka controllers and CRDs
22:54:09 [ OK ] Generating kafka controllers and CRDs
22:54:09 [ .. ] Generating kms controllers and CRDs
22:54:20 [ OK ] Generating kms controllers and CRDs
22:54:20 [ .. ] Generating lambda controllers and CRDs
22:54:35 [ OK ] Generating lambda controllers and CRDs
22:54:35 [ .. ] Generating rds controllers and CRDs
22:54:52 [ OK ] Generating rds controllers and CRDs
22:54:52 [ .. ] Generating secretsmanager controllers and CRDs
22:55:08 [ OK ] Generating secretsmanager controllers and CRDs
22:55:08 [ .. ] Generating servicediscovery controllers and CRDs
22:55:24 [ OK ] Generating servicediscovery controllers and CRDs
22:55:24 [ .. ] Generating sfn controllers and CRDs
22:55:43 [ OK ] Generating sfn controllers and CRDs
22:55:43 [ .. ] verify dependencies have expected content
all modules verified
22:56:00 [ OK ] go modules dependencies verified
22:56:01 [ .. ] go generate darwin_amd64
22:56:34 [ OK ] go generate darwin_amd64
22:56:34 [ .. ] go mod tidy
22:56:36 [ OK ] go mod tidy
22:56:37 [ .. ] cleaning generated CRDs
22:56:38 [ OK ] cleaned generated CRDs
22:56:38 [ .. ] checking that branch is clean
22:56:38 [ OK ] branch is clean
```

**Comment by haarchri** [comment]
@chlunde  thank for mentioned this *trick* in build submodule - is now in place https://github.com/upbound/build/pull/168
for the rest i rebased the branch and cleaned commit history ..

> would be possible to update more of the code generator configs to exclude fields 
think a lot of Naming Changes from new CodeGenerator Version is under the hood CORS vs. Cors / Id vs. ID... 

can you rerun the CI steps ?

File: package/crds/cloudfront.aws.crossplane.io_distributions.yaml [hunk 1]
```diff
@@ -1606,7 +1606,7 @@ spec:
                                         configured as a website endpoint, use the
                                         CustomOriginConfig element instead.
                                       properties:
-                                        originAccessIDentity:
+                                        originAccessIdentity:
```
Comment (negz) [inline package/crds/cloudfront.aws.crossplane.io_distributions.yaml:1609]: Hrm, this seems like a breaking API change. Acceptable for a v1alpha1 resource (and seemingly a bug fix) but we should make note of this in the release notes.

File: package/crds/lambda.aws.crossplane.io_functions.yaml [hunk 2]
```diff
@@ -447,6 +440,21 @@ spec:
...
+                          type: string
+                        codeSize:
+                          format: int64
+                          type: integer
+                        signingJobARN:
+                          type: string
+                        signingProfileVersionARN:
+                          type: string
+                      type: object
+                    type: array
```
Comment (negz) [inline package/crds/lambda.aws.crossplane.io_functions.yaml:457]: I think this is a breaking API change too.

File: pkg/controller/lambda/function/setup.go [hunk 3]
```diff
@@ -439,14 +439,14 @@ func GenerateUpdateFunctionConfigurationInput(cr *svcapitypes.Function) *svcsdk.
...
-		res.SetLayers(f10)
-	}
+	// if cr.Spec.ForProvider.Layers != nil {
+	// 	f10 := []*string{}
+	// 	for _, f10iter := range cr.Spec.ForProvider.Layers {
+	// 		var f10elem = *f10iter
+	// 		f10 = append(f10, &f10elem)
+	// 	}
+	// 	res.SetLayers(f10)
+	// }
```
[the remark this answers is not in the document]
Comment (haarchri) [reply pkg/controller/lambda/function/setup.go]: cleaned the stuff - thanks for hint

File: package/crds/cloudfront.aws.crossplane.io_distributions.yaml (hunk 1 above)
Comment (haarchri) [reply package/crds/cloudfront.aws.crossplane.io_distributions.yaml:1609]: tested the resource via example and changed the field in the example - right we need to mention this issue as breaking API change

File: package/crds/lambda.aws.crossplane.io_functions.yaml (hunk 2 above)
Comment (haarchri) [reply package/crds/lambda.aws.crossplane.io_functions.yaml:457]: it is only in status filed - so i think this is no breaking api change

**Comment by haarchri** [comment]
we will follow up with https://github.com/crossplane/provider-aws/pull/918

**Pull request closed without merging**